core/app/models/spree/stock_item.rb

```diff
@@ -96,12 +96,19 @@ def process_backorders(number)
       end
 
       def conditional_variant_touch
-        # the variant_id changes from nil when a new stock location is added
-        stock_changed = (count_on_hand_changed? && count_on_hand_change.any? { |c| c < 1 }) || variant_id_changed?
-
-        if !Spree::Config.binary_inventory_cache || stock_changed
-          variant.touch
+        # only warn if store is setting binary_inventory_cache (default = false)
+        if Spree::Config.binary_inventory_cache
+          Spree::Config.inventory_cache_threshold = 1
```

Comment:
I like that approach @jhawthorn. I might even take it one step further and define an override to the inventory_cache_threshold on Spree::Config to defer to binary_inventory_cache if set to true and issue the warning there, but both seem good